### PR TITLE
[CSL-758] Update downloading: improvements and refactorings

### PR DESCRIPTION
**Pull request by flyingleafe**

### CODE REVIEW

File: src/Pos/Update/Download.hs [hunk 1]
```diff
@@ -33,34 +35,50 @@ import           Pos.WorkMode            (WorkMode)
...
+        updHash <- maybe (throwError "This update is not for our system")
+                   pure mupdHash
+        let updateVersion = upSoftwareVersion cpsUpdateProposal
+        unless (versionIsNew updateVersion) $
+            throwError $ sformat ("Update #"%build%" hasn't been downloaded: \
+                                  \current software version is newer than \
+                                  \update version") updHash
+
+        updPath <- npUpdatePath . ncNodeParams <$> getNodeContext
+        -- let updAppName = svAppName . upSoftwareVersion $
```
Comment (volhovM) [inline src/Pos/Update/Download.hs]: Delete this one please.

File: src/Pos/Update/Download.hs [hunk 2]
```diff
@@ -33,34 +35,50 @@ import           Pos.WorkMode            (WorkMode)
 showHash :: Hash a -> FilePath
 showHash = toString . B16.encode . BA.convert
 
+-- TODO: if we're downloading update not for `cardano-sl`,
+-- but e. g. for `daedalus`, how do we check that version is new?
```
Comment (volhovM) [inline src/Pos/Update/Download.hs]: Huh, that's a pretty good question indeed. I suppose that current check is correct and application name is "backend node" name, so there's no software app name "daedalus". Though i may be wrong. cc @georgeee

**Comment by volhovM** [comment]
Since CI doesn't work now, i've checked manually: it builds, tests pass (except for tests that were broken before).

File: src/Pos/Update/Download.hs (hunk 2 above)
Comment (flyingleafe) [inline src/Pos/Update/Download.hs]: I think that for now this should work correctly - we probably won't have updates not for `cardano-sl` for a while (if we will have them at all). Probably the good solution would be to create an issue about that and put it into backlog.

**Pull request merged**